Add theme toggle, dark mode support, and accessibility enhancements w/ semantic html

Clonephaze opened the pull request:
Introduces a theme toggle in the Drawer component, allowing users to switch between light, dark, and system themes. Refactors SCSS files to use CSS custom properties for colors, enabling dynamic theming and improved dark mode support throughout the UI.

Clonephaze committed:
Add theme toggle and refactor for dark mode support

Introduces a theme toggle in the Drawer component, allowing users to switch between light, dark, and system themes. Refactors SCSS files to use CSS custom properties for colors, enabling dynamic theming and improved dark mode support throughout the UI.

ewauq commented:
Thank you very much for your contribution, it's truly appreciated! I'll review it very soon.

Clonephaze commented on src/component/Drawer.scss:
```diff
@@ -139,3 +167,14 @@
+    &:hover {
+      background-color: #e6c800;
+    }
+  }
```
Valid, I'll do this really quick

Clonephaze commented on src/component/Drawer.tsx:
```diff
@@ -51,6 +93,11 @@ const Drawer = (): JSX.Element => {
+        <a className="theme-toggle" onClick={cycleTheme}>
+          <span className="theme-label">Theme</span>
+          <span className="theme-value">{getThemeIcon()} {getThemeLabel()}</span>
+        </a>
```
Fixing with the accessibility PR

Clonephaze commented on src/style/theme.scss:
```diff
@@ -25,22 +25,134 @@ $color-green: rgb(126, 196, 136);
+    --link-color: light-dark(#{$color-blue-link}, #7ab3ff);
+    --icon-invert: light-dark(0, 0.85);
+  }
+}
```
I'm not sure what copilot means here, light-dark absolutely works in modern browsers. Has for a while now.

Clonephaze commented on src/style/theme.scss:
```diff
@@ -25,22 +25,134 @@ $color-green: rgb(126, 196, 136);
+:root {
+  // Fallback for browsers that don't support light-dark()
+  --bg-primary: #{$light-bg-primary};
+  --bg-secondary: #{$light-bg-secondary};
```
I left it as an option, I'll leave that up to you.

Clonephaze committed:
Improve dark mode support for support link in Drawer

Added explicit dark mode background styles for the 'buy me a coffee' support link in Drawer.scss. Also enhanced Drawer.tsx with JSDoc comments for theme-related functions to improve code clarity.

Clonephaze committed:
Improve accessibility and semantics across UI components

Refactors multiple components to enhance accessibility, including adding ARIA attributes, semantic HTML elements (header, main, nav), and keyboard focus styles. Introduces a skip-to-content link, updates buttons and links with appropriate roles and labels, and ensures external links use rel="noopener noreferrer". Also restructures navigation lists for better semantics and updates styles to support these changes.

Clonephaze commented:
I just went ahead and put those accessibility changes in here too.

Clonephaze committed:
Improve accessibility for Line checkbox

Replaces the checkbox's aria-label and visually hidden label with aria-labelledby referencing the text span, ensuring better screen reader support and simplifying the markup.

Clonephaze commented on src/component/Line.tsx:
```diff
@@ -29,10 +29,14 @@ const Line = ({ pageId, id, text }: LineProps): JSX.Element => {
       />
+      <label htmlFor={`chapter-${pageId}-item-${id}`} className="sr-only">
+        Mark as complete
+      </label>
```
This particular line? It allows screen reader users to much more easily tell which checkbox they have selected. But I just enhanced this step to make it more clear for those users, by using the text for their associated line be registered as the checkbox label. So a screen reader user will hear "Defeat the Eye of Cthulhu, checkbox".

Clonephaze commented:
If there's anything you don't like about the accessibility changes or would rather I walk them back from this pr just let me know.

ewauq committed:
Update the Party Girl requirements

ewauq committed:
Update footer infos

ewauq committed:
Fix the icon color on dark mode

ewauq committed:
Fix icons colors on auto mode

ewauq committed:
Remove old dark mode switch

ewauq committed:
Update strips for dark mode

ewauq committed:
Update dark mode style

ewauq committed:
Remove unused assets

ewauq committed:
Update footer credits

ewauq committed:
Update dark mode styles

ewauq committed:
Update dark mode switch style and remove home button

ewauq committed:
Update actions icons
